### PR TITLE
Add pyrser

### PR DESCRIPTION
**Pull request by psortos**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/treematching/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+test:
+  imports:
+    - treematching
+
+about:
+  home: https://github.com/LionelAuroux/treematching
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "Pyrser is a toolbox with : A PEG parser, AST rewriting helpers and Type System."
```
Comment (dopplershift) [inline recipes/treematching/meta.yaml:33]: Copy/paste oops?

File: recipes/treematching/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - python >=3.6
+
+test:
+  imports:
+    - treematching
+
+about:
+  home: https://github.com/LionelAuroux/treematching
+  license: MIT
+  license_family: MIT
```
Comment (dopplershift) [inline recipes/treematching/meta.yaml:31]:
```suggestion
  license: GPL-2.0-Only
  license_family: GPL
```

File: recipes/pyrser/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{% set name = "pyrser" %}
+{% set version = "2.5.0" %}
```
Comment (dopplershift) [inline recipes/pyrser/meta.yaml:2]: Where is this version coming from? PyPI has 0.2 and that's what I see in upstream?

File: recipes/pyrser/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    - pprintpp
+    - jinja2
+test:
+  imports:
+    - pyrser
+
+about:
+  home: https://github.com/LionelAuroux/pyrser
+  license: MIT
+  license_family: MIT
```
Comment (dopplershift) [inline recipes/pyrser/meta.yaml:34]:
```suggestion
  license: GPL-3.0-only
  license_family: GPL
```

File: recipes/pyrser/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - python >=3.6
+    - treematching
```
Comment (dopplershift) [inline recipes/pyrser/meta.yaml:23]: How is this a dependency if it's not on PyPI?

File: recipes/pyrser/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{% set name = "pyrser" %}
+{% set version = "2.5.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/LionelAuroux/{{ name }}/archive/master.tar.gz
```
Comment (dopplershift) [inline recipes/pyrser/meta.yaml:9]: This needs to not just be a snapshot off master, it really needs to be some kind of tagged version.

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/pyrser/meta.yaml (hunk 5 above)
Comment (psortos) [reply recipes/pyrser/meta.yaml]: That was a mistake, removed!

File: recipes/pyrser/meta.yaml (hunk 6 above)
Comment (psortos) [reply recipes/pyrser/meta.yaml]: Switched it to use the wheel file found in pypi https://pypi.org/project/pyrser/#files

File: recipes/treematching/meta.yaml (hunk 1 above)
Comment (psortos) [reply recipes/treematching/meta.yaml]: Replaced!

File: recipes/pyrser/meta.yaml (hunk 3 above)
Comment (psortos) [reply recipes/pyrser/meta.yaml]: updated this to 0.2.0

**Review by psortos — COMMENTED** [review]
Updated the recipes accordingly, they are each passing all checks

File: recipes/treematching/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{% set name = "treematching" %}
+{% set version = "0.0.1" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/LionelAuroux/{{ name }}/archive/master.tar.gz
```
Comment (dopplershift) [inline recipes/treematching/meta.yaml:9]: We really need some kind of release to tie this to.

File: recipes/pyrser/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+{% set name = "pyrser" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/py2.py3/p/{{ name }}/{{ name }}-{{ version }}-py2.py3-none-any.whl
```
Comment (dopplershift) [inline recipes/pyrser/meta.yaml:9]: Installing from wheels is really not a great way to do this. Is there any way to get an sdist/release tarball from upstream?

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/pyrser/meta.yaml (hunk 8 above)
Comment (psortos) [reply recipes/pyrser/meta.yaml:9]: There are no current release tars for this repository, but I submitted an issue for this repo to add one.
 I used this previous PR as a reference https://github.com/conda-forge/tensorflow-metadata-feedstock/blob/master/recipe/meta.yaml

Comment (oblute) [reply recipes/pyrser/meta.yaml:9]: The only way to do it currently is via a wheel (there are no tarballs). We can make an issue on the repo and include a comment in the recipe that we will change if possible. A recent PR from a teammate was just merged, with a wheel, at the encouragement of a staged-recipes maintainer: https://github.com/conda-forge/staged-recipes/pull/14610#issuecomment-831491197

**Comment by oblute** [comment]
@conda-forge/staged-recipes @dopplershift ready for final review/merge

File: recipes/treematching/meta.yaml (hunk 7 above)
Comment (xhochy) [reply recipes/treematching/meta.yaml:9]: Installing from master is a blocker for this, we cannot package this as such.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by xhochy** [comment]
Have a try with `treematching` in a separate PR using a different target channel and date-based versioning as we did for the `asv` nightlies: https://github.com/conda-forge/asv-feedstock/tree/dev/recipe

**Comment by psortos** [comment]
@xhochy just pyrser, removed treematching

**Pull request merged**